src/AzurePipelinesToGitHubActionsConverter.Core/Conversion.cs

```diff
@@ -48,38 +48,39 @@ private ConversionResponse ConvertAzurePipelineToGitHubActionV2(string yaml)
             List<string> stepComments = new List<string>();
 
             //convert the yaml into json, it's easier to parse
-            JObject json = null;
+            JsonElement jsonObject = new JsonElement();// = null;
             if (yaml != null)
             {
                 //Clean up the YAML to remove conditional insert statements
                 string processedYaml = ConversionUtility.CleanYamlBeforeDeserializationV2(yaml);
-                json = JsonSerialization.DeserializeStringToObject(processedYaml);
+                jsonObject = JsonSerialization.DeserializeStringToObject(processedYaml);
             }
 
             //Build up the GitHub object piece by piece
             GitHubActionsRoot gitHubActions = new GitHubActionsRoot();
             GeneralProcessing gp = new GeneralProcessing(_verbose);
 
-            if (json != null)
+            if (jsonObject.ValueKind != JsonValueKind.Undefined)
             {
+                JsonElement jsonElement = new JsonElement();
                 //Name
-                if (json["name"] != null)
+                if (jsonObject.TryGetProperty("name", out jsonElement) == true)
```

Comment:
Remember to double check all of the out variables - I think I can simplify most of the local variables